src/test/spec/command_monitoring/operation.rs

```diff
@@ -267,18 +281,39 @@ pub(super) struct UpdateOne {
     upsert: Option<bool>,
 }
 
+#[async_trait]
 impl TestOperation for UpdateOne {
     fn command_names(&self) -> &[&str] {
         &["update"]
     }
 
-    fn execute(&self, collection: Collection) -> Result<()> {
+    async fn execute(&self, collection: Collection) -> Result<()> {
         let options = self.upsert.map(|b| UpdateOptions {
             upsert: Some(b),
             ..Default::default()
         });
         collection
             .update_one(self.filter.clone(), self.update.clone(), options)
+            .await
             .map(|_| ())
     }
 }
+
+impl EventClient {
+    pub(super) async fn run_operation_with_events(
+        &self,
+        operation: AnyTestOperation,
+        database_name: &str,
+        collection_name: &str,
+    ) -> Vec<CommandEvent> {
+        let _: Result<_> = operation
+            .execute(self.database(database_name).collection(collection_name))
+            .await;
+        self.command_events
+            .write()
```

Comment:
All these events still use regular sync locks. I experimented with updating the event handlers to fire off background tasks for populating the event lists, but it was too nondeterministic for testing purposes. Because we're using a different client for each test nowadays, I think the risk for deadlock is minimal, so I decided to just leave all the tests' sync event handling code alone until we decide something else. Filed [RUST-340](https://jira.mongodb.org/browse/RUST-340) to discuss options there.